pandas/core/config_init.py

```diff
@@ -279,7 +280,8 @@ def mpl_style_cb(key):
 
 with cf.config_prefix('display'):
     cf.register_option('precision', 6, pc_precision_doc, validator=is_int)
-    cf.register_option('float_format', None, float_format_doc)
+    cf.register_option('float_format', None, float_format_doc,
+                       validator=is_callable)
```

Comment:
Looks like `float_format = None` is handled [here](https://github.com/tdhopper/pandas/blob/64950117a2d09ea3750c76235282079640782bc5/pandas/core/format.py#L2031-L2033).